src/test/java/Hardcore/page/GoogleCloudPricingCalculatorPage.java

```diff
@@ -0,0 +1,142 @@
+    @FindBy(xpath = "//md-select[@placeholder='GPU type']")
+    WebElement dropboxGPUType;
+    @FindBy(xpath = "//md-option[@value='NVIDIA_TESLA_V100']")
+    WebElement itemGPUType;
+
+    @FindBy(xpath = "//md-select[@placeholder='Local SSD']")
+    WebElement dropdownSSD;
+    @FindBy(xpath = "//md-option[@value='2' and contains(@ng-repeat, 'item in listingCtrl.dynamicSsd.computeServer')]")
+    WebElement itemSSD;
+
+    @FindBy(xpath = "//md-select[@placeholder='Datacenter location'][@ng-model='listingCtrl.computeServer.location']")
+    WebElement dropdownLocation;
+    @FindBy(xpath = "//md-select-menu[@class='md-overflow']/descendant::md-option[@value='europe-west3' and contains(@ng-repeat, 'item in listingCtrl.fullRegionList')]")
+    WebElement itemLocation;
+
+    @FindBy(xpath = "//md-select[@placeholder='Committed usage']")
+    WebElement dropdownUsage;
+    @FindBy(xpath = "//div[@class='md-select-menu-container md-active md-clickable']/descendant::md-option[@ng-value='1'][@value='1']")
+    WebElement itemUsage;
+
+    @FindBy(xpath = "//button[@aria-label='Add to Estimate' and contains(@ng-click, 'addComputeServer')]")
+    WebElement buttonAddToEstimate;
+
+    @FindBy(xpath = "//h2/b[@class='ng-binding']")
+    WebElement textTotalCost;
+
+    @FindBy(xpath = "//button[@aria-label='Email Estimate']")
+    WebElement buttonEmailEstimate;
+
+    @FindBy(xpath = "//input[@ng-model='emailQuote.user.email']")
+    WebElement inputEmail;
+
+    @FindBy(xpath = "//button[@aria-label='Send Email']")
+    WebElement buttonSendEmail;
+
+    public GoogleCloudPricingCalculatorPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public GoogleCloudPricingCalculatorPage switchToCalculatorIFrame() {
+        driver.switchTo().frame(frameMain);
+        driver.switchTo().frame("myFrame");
+        return this;
+    }
+
+    public GoogleCloudPricingCalculatorPage fillInEstimationForm() {
+        String instances = "4";
+
+        waitForWebElementVisible(buttonComputeEngine).click();
+        inputNumberOfInstances.sendKeys(instances);
+        inputPurpose.clear();
+        dropdownSoftware.click();
+        waitForWebElementVisible(itemSoftware).click();
+        dropdownClass.click();
+        waitForWebElementVisible(itemClass).click();
+        dropdownSeries.click();
+        waitForWebElementVisible(itemSeries).click();
+        dropdownMachineType.click();
+        waitForWebElementVisible(itemMachineType).click();
+        checkboxAddGPUs.click();
+        waitForWebElementVisible(dropboxNumberOfGPUs).click();
+        waitForWebElementVisible(itemNumberOfGPUs).click();
+        dropboxGPUType.click();
+        waitForWebElementVisible(itemGPUType).click();
+        dropdownSSD.click();
+        waitForWebElementVisible(itemSSD).click();
+        dropdownLocation.click();
+        waitForWebElementVisible(itemLocation).click();
+        dropdownUsage.click();
+        waitForWebElementVisible(itemUsage).click();
```

Comment:
The method is overloaded and not clear - need to decompose it - you can split it into smaller logical private methods and trigger them here in the public method.
Moreover private small methods can be converted into public and used in your tests when they need to be extended